### PR TITLE
Hard 'fork' the solution crawler code.  Placing the fork in the unit-testing EA layer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/64147

This creates a full copy of all the code necessary for SolutionCrawler, and places it in the unit testing EA layer.  Important notes:

1. None of this code has, or is allowed to add, a reference to Microsoft.CodeAnalysis.SolutionCrawler.  That namespace+types are going to go away entirely, so this fork must be independently compilable.
2. Currently, nothing starts this crawler.  The registration hooks are not activated by any existing code in roslyn, and there is no equivalent to HostSolutionCrawlerWorkspaceEventListener which is listening to the workspace and activating this.  This will be activated in a future PR by the new 'ILegacySolutionEvent' component when there is a listener that UnitTesting exports.
3. *All* the types have had 'UnitTesting' added to their names to make it clear that this is only for that purpose, and to prevent confusion with the coexisting types in the workspace today.
4. Other than the naming, the move to hte new namespace, and removal of any 'using' of hte old namespace, the code was not touched *at all*.  The intent is to keep the logic untouched in this PR so that future logic changes are trivial to review.

There will be many more PRs coming on this.  I'm trying ot break these up into manageable chunks though to keep the overall design and individual steps clear.  This is all going into a feature branch as well until ready to actually merge in.

### CODE REVIEW

File: src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs:1]: Going in with https://github.com/dotnet/roslyn/pull/64147

File: src/EditorFeatures/Core/SolutionEvents/LegacySolutionEvent.cs [hunk 2]
```diff
@@ -4,17 +4,17 @@
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/SolutionEvents/LegacySolutionEvent.cs:4]: Going in with https://github.com/dotnet/roslyn/pull/64147

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/AbstractUnitTestingDocumentDifferenceService.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/AbstractUnitTestingDocumentDifferenceService.cs:1]: Everything in this folder is hte hard fork, with the only changes being that *ALL* the type names have all had 'UnitTesting' added to them.

File: src/Workspaces/Remote/ServiceHub/Services/LegacySolutionEvents/RemoteLegacySolutionEventsAggregationService.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/LegacySolutionEvents/RemoteLegacySolutionEventsAggregationService.cs:1]: Going in with https://github.com/dotnet/roslyn/pull/64147

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
@shyamnamboodiripad to keep you up to date on what we're doing here.

**Pull request merged**